idl/internal/thrift.y

```diff
@@ -386,13 +386,12 @@ base_type_name
 /***************************************************************************
  Constant values
  ***************************************************************************/
-
 const_value
-    : INTCONSTANT { $$ = ast.ConstantInteger($1) }
-    | DUBCONSTANT { $$ = ast.ConstantDouble($1) }
-    | TRUE        { $$ = ast.ConstantBoolean(true) }
-    | FALSE       { $$ = ast.ConstantBoolean(false) }
-    | LITERAL     { $$ = ast.ConstantString($1) }
+    : INTCONSTANT { $$ = ast.ConstantInteger($1); yylex.(*lexer).RecordPosition($$) }
+    | DUBCONSTANT { $$ = ast.ConstantDouble($1); yylex.(*lexer).RecordPosition($$) }
+    | TRUE        { $$ = ast.ConstantBoolean(true); yylex.(*lexer).RecordPosition($$) }
+    | FALSE       { $$ = ast.ConstantBoolean(false); yylex.(*lexer).RecordPosition($$) }
+    | LITERAL     { $$ = ast.ConstantString($1); yylex.(*lexer).RecordPosition($$) }
```

Comment:
I suspect this will record the line number right *after* the token.

This is probably not an issue for constant primitives because they'll never be split across multiple lines, but if we want accurate column numbers later, the position would have to be recorded before the value (like with `lineno IDENTIFIER` below). I'm okay doing that when we add support for column numbers, though.